### PR TITLE
Fix serialization for deleted trajectories.

### PR DESCRIPTION
**Pull request by sebastianklose**
* update TrajectoryState of trajectories that got 'trimmed' away to be
 deleted in the PureLocalizationTrimmer
 * update serialization to only serialize 'undeleted' trajectories and
 corresponding options.

#1111

### CODE REVIEW

File: cartographer/io/internal/mapping_state_serialization.cc [hunk 1]
```diff
@@ -197,8 +212,9 @@ void WritePbStream(
     ProtoStreamWriterInterface* const writer) {
   writer->WriteProto(CreateHeader());
   writer->WriteProto(SerializePoseGraph(pose_graph));
-  writer->WriteProto(
-      SerializeAllTrajectoryBuilderOptions(trajectory_builder_options));
+  writer->WriteProto(SerializeAllTrajectoryBuilderOptions(
```
Comment (gaschler) [inline cartographer/io/internal/mapping_state_serialization.cc]: Perhaps change remove the "all" from the function name because it's no longer the case?

Comment (sebastianklose) [reply cartographer/io/internal/mapping_state_serialization.cc]: good catch :-)

**Comment by sebastianklose** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user sebastianklose. Merge queue now has a length of 1.

**Pull request merged**